src/formatDate.js

```diff
@@ -8,7 +8,44 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitSeparator = fromFormat[fromFormat.length - 1];
+  const splitDate = date.split(splitSeparator);
+
+  const joinSeparator = toFormat[toFormat.length - 1];
+
+  // variables of index date fromFormate
+  const fromYYYY = fromFormat.includes('YY')
+    ? fromFormat.indexOf('YY')
+    : fromFormat.indexOf('YYYY');
+  const fromMM = fromFormat.indexOf('MM');
+  const fromDD = fromFormat.indexOf('DD');
+
+  // variables of index date toFormate
+  const toYYYY = toFormat.includes('YY')
+    ? toFormat.indexOf('YY')
+    : toFormat.indexOf('YYYY');
+  const toMM = toFormat.indexOf('MM');
+  const toDD = toFormat.indexOf('DD');
+
+  if (fromFormat[fromYYYY].length > toFormat[toYYYY].length) {
+    splitDate[fromYYYY] = splitDate[fromYYYY].slice(2);
+  }
```

Comment:
This block correctly handles the conversion from a four-digit year to a two-digit year. Ensure that the input year is always valid to prevent slicing errors.